fusequery/query/src/servers/mysql/mysql_interactive_worker.rs

```diff
@@ -93,7 +93,14 @@ impl<W: std::io::Write> MysqlShim<W> for InteractiveWorker<W> {
         let context = self.session.create_context();
 
         context.attach_query_info(query);
-        DFQueryResultWriter::create(writer).write(self.base.do_query(query, context))?;
+        match DFQueryResultWriter::create(writer).write(self.base.do_query(query, context)) {
+            Ok(_) => {}
+            Err(e) => {
+                let query_str = String::from(query);
+                let new_error = e.add_message(query_str);
```

Comment:
Hi,
 let new_error = e.add_message(query_str.to_string()); ?

Reply:
```
pub fn add_message(self, msg: String)

--> 


pub fn add_message(self, msg: AsRef<str>)

```

Maybe this is better?

Reply:
> ```
> pub fn add_message(self, msg: String)
> 
> --> 
> 
> 
> pub fn add_message(self, msg: impl AsRef<str>)
> ```
> 
> Maybe this is better?

:+1:

Reply:
> ```
> pub fn add_message(self, msg: String)
> 
> --> 
> 
> 
> pub fn add_message(self, msg: impl AsRef<str>)
> ```
> 
> Maybe this is better?

yes sir, I will try to modify.